packages/autocomplete-core/src/__tests__/concurrency.test.ts

```diff
@@ -0,0 +1,58 @@
+import userEvent from '@testing-library/user-event';
+
+import { createAutocomplete } from '../createAutocomplete';
+
+function defer<TValue>(fn: () => TValue, timeout: number) {
+  return new Promise<TValue>((resolve) => {
+    setTimeout(() => resolve(fn()), timeout);
+  });
+}
+
+describe('concurrency', () => {
+  test('resolves the responses in order from getSources', async () => {
+    // These delays make the second query come back after the third one.
+    const delays = [100, 300, 200];
+    let deferCount = -1;
+
+    const getSources = ({ query }) => {
+      deferCount++;
+
+      return defer(() => {
+        return [
+          {
+            getItems() {
+              return [{ label: query }];
+            },
+          },
+        ];
+      }, delays[deferCount]);
+    };
+    const onStateChange = jest.fn();
+    const autocomplete = createAutocomplete({ getSources, onStateChange });
+    const { onChange } = autocomplete.getInputProps({ inputElement: null });
+    const input = document.createElement('input');
+    input.addEventListener('input', onChange);
+    document.body.appendChild(input);
```

Comment:
I'd move this setup code of the test block for clarity, just so it's easier to spot the act and assert part.

Reply:
Since there's no guidelines about this yet and it's based on personal preference. I prefer keeping it contained in the test for now.